oml/models/vit_unicom/extractor.py

```diff
@@ -0,0 +1,95 @@
+from pathlib import Path
+from typing import Optional, Union
+
+import torch
+
+from oml.const import CKPT_SAVE_ROOT
+from oml.interfaces.models import IExtractor
+from oml.models.utils import remove_prefix_from_state_dict
+from oml.models.vit_unicom.external import vision_transformer
+from oml.models.vit_unicom.external.model import load  # type: ignore
+from oml.utils.misc_torch import normalise
+
+
+def unicom_vitb32() -> vision_transformer.VisionTransformer:  # type: ignore
+    return vision_transformer.build_model("ViT-B/32")  # type: ignore
+
+
+def unicom_vitb16() -> vision_transformer.VisionTransformer:  # type: ignore
+    return vision_transformer.build_model("ViT-B/16")  # type: ignore
+
+
+def unicom_vitl14() -> vision_transformer.VisionTransformer:  # type: ignore
+    return vision_transformer.build_model("ViT-L/14")  # type: ignore
+
+
+def unicom_vitl14_336px() -> vision_transformer.VisionTransformer:  # type: ignore
+    return vision_transformer.build_model("ViT-L/14@336px")  # type: ignore
+
+
+class ViTUnicomExtractor(IExtractor):
+    constructors = {
+        "vitb32_unicom": unicom_vitb32,
+        "vitb16_unicom": unicom_vitb16,
+        "vitl14_unicom": unicom_vitl14,
+        "vitl14_336px_unicom": unicom_vitl14_336px,
+    }
+
+    pretrained_models = {
+        "vitb32_unicom": {
+            "download_fn": lambda: load("ViT-B/32", download_root=CKPT_SAVE_ROOT),
+            "fname": "FP16-ViT-B-32.pt",
```

Comment:
You don't use `fname` key as well as `IExtractor`

Reply:
Probably no one checks files in `.cache`. Ok, I check, but not users

Reply:
thx!